### DISCUSSION TITLE
3D Shapes Following the Earth's Curvature

### DISCUSSION THREAD

**Discussion by nebsar** [pelicanmapping/osgearth#2039] [Support Q&A]
Is it possible to render 3D airspaces or shapes following the earth's curvature, like in the image below?

![image](https://user-images.githubusercontent.com/5169314/162075193-2dc386de-1e9e-4c98-beba-1fbb1212734e.png)

**Comment by osgwraiESC** [comment]
see osgearth_annotation/osgearth_annotation.cpp, that will likely help you

**Comment by nebsar** [comment]
I know that example. How about the minimum altitude of the shape? Is it the base altitude of the shape, just like in below code?

Style ellipseStyle;
		ellipseStyle.getOrCreate<PolygonSymbol>()->fill()->color() = Color(Color::Blue, 0.75);
		ellipseStyle.getOrCreate<ExtrusionSymbol>()->height() = 250000.0; // meters MSL
		EllipseNode* ellipse = new EllipseNode();
        ellipse->set(
			GeoPoint(geoSRS, -80.28, 25.82, 0.0, ALTMODE_RELATIVE), // IS THIS THE BASE ALTITUDE?
			Distance(250, Units::MILES),
			Distance(100, Units::MILES),
			Angle   (0, Units::DEGREES),
			ellipseStyle,
            Angle(-40.0, Units::DEGREES),
            Angle(40.0, Units::DEGREES),
            true);

**Comment by osgwraiESC** [comment]
not exactly sure how it will look differently if you use 0 altitude or the exact terrain altitude at your lat/ lon points ... when i used it had the lat, lon, altitude of mouse points from the mouse coords tool, added extrustion and altitude symbols like in the examples in that file

**Reply by nebsar** [reply]
Ok. I would like to add 3d airspaces which have a base altitude higher than 0 feet.

**Reply by osgwraiESC** [reply]
to create this i used the local terrain height plus some delta altitude

![corridor1](https://user-images.githubusercontent.com/102617369/162234686-9dccb4aa-1507-4797-9a6d-93c13069bf8d.png)

**Reply by nebsar** [reply]
Ok, it is good to know. Thank you.
